internal/chain/state.go

```diff
@@ -39,40 +39,33 @@ type storeState struct {
 	record   state.Chain
 }
 
-// NewStateManager creates a new state manager and loads the transaction's
-// sponsor. If the sponsor is not found, NewStateManager returns a valid state
-// manager along with a not-found error.
-func NewStateManager(db *state.StateDB, tx *transactions.GenTransaction) (*StateManager, error) {
+func NewStateManager(dbTx *state.DBTransactional, tx *transactions.GenTransaction) (*StateManager, error) {
 	m := new(StateManager)
-	m.db = db
+	m.dbTx = dbTx
 	m.chains = map[[32]byte]state.Chain{}
 	m.stores = map[[32]byte]*storeState{}
 	m.writes = map[storage.Key][]byte{}
 	m.txHash = types.Bytes(tx.TransactionHash()).AsBytes32()
 	m.txType = tx.TransactionType()
 
-	// The genesis TX is special
```

Comment:
I think you need to merge with `develop`. Your changes are reverting some of the comments I added.